### ISSUE TITLE
Science to Software

### ISSUE THREAD

**Issue by cthoyt** [jcheminform/jcheminform#13]
# Science to Software

This template can be used to propose a collection in the Journal of Cheminformatics (see [this list of current and past collections](https://jcheminf.biomedcentral.com/articles/collections)). A collection is also known as a special issue or thematic series, and is used to group articles about a certain theme (like [PubChem 3D](https://www.biomedcentral.com/collections/pubchem3d)) or around a conferences (like [the proceedings of the 11th ICCS](https://www.biomedcentral.com/collections/ICCS11)). This issue allows us to coordinate the discussion and development of a collection proposal. A collection often is linked to a [Call of Papers](https://jcheminf.biomedcentral.com/upcoming-special-issues). The main text of this ticket can be edited after you submit it, as more details become clear, or when updates are made based on discussions. The first post is just to start the discussion. You can remove this paragraph before you submit your proposal.

## Checklist

You acknowledge the [submission guidelines](https://jcheminf.biomedcentral.com/submission-guidelines) and in particular that all contributions adhere to the regular journal expectations:

 - [ ] source code and scripts with an OSI-approved license
 - [ ] data is provided under an open data license (CCZero, public domain, CC-BY)

## Proposal

For many chemoinformaticians, there's a tension between writing research code and applying software engineering practices that make code reproducible, reusable, and, ultimately, useful for others. It arises due to several circumstances often including time pressure, lack of training, lack of motivation, and lack of community enforcement/standards. However, these are not natural opponents. There are a few steps that can be taken to turn any research code into "software" that we would like to highlight in a J Chem. Inf. collection. A few goals would be to:

1. Demystify the ivory tower of "software" and "software engineering" - it's not meant to be a closed club of people with computer science degrees or people living in Silicon Valley but rather something we can all do (new short submissions)
   - This must include an obligatory discussion about the difference between reproducibility and open source code, and why the phrase FAIR doesn't exactly apply to software the same way as data (though programs like [pyroma](https://pypi.org/project/pyroma/) for Python are a great analog)
2. Examples of articles in J Chem. Inf. that go above and beyond simply dumping code to GitHub, but have also packaged the code for reuse and make effort to explain how their code was used to make the presentation of the paper (previous submissions)
   - https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00536-w
3. The case for why _you_ should do it too
   - Discussion of PyBacting experience?
4. Practical tutorials on how to package code in several languages (Python, R, Java, etc.) as well as distribute through community channels (PyPI and Conda Forge for Python, CRAN and BioConductor for R, Maven and Gradle for Java, etc.) (new short submissions)
5. Examples of turning previous publications into reusable code
    - Pat Walter's clusterama (https://github.com/cthoyt-forks-and-packages/clusterama)
    - AstraZenea's nonadditivity analysis (https://github.com/cthoyt/AZNonadditivityAnalysis)

 - [x] this collection is about a certain theme
 - [ ] this collection is for a conference

## Keywords

reproducibility, reusability, open science

## Proposed guest editors

 * Charles Tapley Hoyt (Harvard Medical School) (@cthoyt, [orcid:0000-0003-4423-4370](https://bioregistry.io/orcid:0000-0003-4423-4370), [wikidata:Q47475003](https://bioregistry.io/wikidata:Q47475003), [twitter:@cthoyt](https://twitter.com/cthoyt), https://cthoyt.com; assuming it's not bad form for me to recommend myself 😉 )
 * Jessica A Nash (The Molecular Sciences Software Institute) ([orcid:0000-0003-1967-5094](https://bioregistry.io/orcid:0000-0003-1967-5094), [twitter:@jessica_a_nash](https://twitter.com/jessica_a_nash)) or perhaps another person from MSSI

**Comment by BZdrazil** [comment]
Sounds like a good idea. Point 4 could be the topic for a thematic collection were you ask for educational articels / tutorials.

**Comment by cthoyt** [comment]
@BZdrazil do you think it would be better to just focus on point 4/5, then maybe have a foreword from the guest editors that touches on the other points?

**Comment by BZdrazil** [comment]
In your description of the special collection you can describe what factors are important to be considered when writing an article including code. But I just meant that the most actionable call for papers might be around new tutorials (that follow your guidelines). For point 5, I am not sure this is something that could be published as a stand-alone paper. If it was developed by others originally and then transformed into something more useful and resueable that's good but might lead to authorship claims by the original authors. I think also the novelty aspect is not so clear in this case. But others might have another opinion.

**Comment by rajarshi** [comment]
I agree with @BZdrazil, sounds like  4 & 5 are the main topics about which contributors could write about, with the caveat about authorship claims as she points out (which can easily be resolved by pull in the authors of the original code as authors on the submission :)

I think points 1-3 would make for an editorial or an article introducing the collection.

Do you have an idea of potential authors? One person I'd suggest is Zach Charlop Powers (@zachcp), who has been instrumental in packaging the rcdk codebase

**Comment by vedina** [comment]
I agree this is a needed topic and may help improve the quality of source code published in the journal,  but it seems to me the option 4) is what makes sense to be considered, and it would fall into the existing Educational type of papers.
